spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume_spec.rb

```diff
@@ -0,0 +1,81 @@
+describe ManageIQ::Providers::IbmCloud::PowerVirtualServers::StorageManager::CloudVolume do
+  let(:ems_cloud) { FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_cloud) }
+  let(:ems_storage) { FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_storage, :parent_ems_id => ems_cloud.id) }
+  let(:cloud_volume) { FactoryBot.create(:cloud_volume_ibm_cloud_power_virtual_servers, :ext_management_system => ems_storage) }
+
+  describe "#validate_delete_volume" do
+    it "validates volume delete when in-use" do
+      cloud_volume.update(:status => "in-use")
+      expect(cloud_volume.validate_delete_volume[:available]).to be_falsy
+    end
+
+    it "validates volume delete when available" do
+      cloud_volume.update(:status => "available")
+      expect(cloud_volume.validate_delete_volume[:available]).to be_truthy
+    end
+
+    it "validates volume delete when ems is missing" do
+      expect(cloud_volume).to receive(:ext_management_system).and_return(nil)
```

Comment:
ext_management_system is a proper association so you shouldn't stub it, just create a context like "when ems is missing" and `let(:ems_storage) { nil }`

Reply:
Done.